packages/sdk/test/methods.spec.ts

```diff
@@ -182,6 +206,59 @@ describe('LookerNodeSDK', () => {
     await sdk.authSession.logout()
   }
 
+  describe('downloads', () => {
+    it('png and svg', async () => {
+      const sdk = new LookerSDK(session)
+      const looks = await sdk.ok(sdk.search_looks({ limit: 1 }))
+      let type = ''
+      let id = ''
+      expect(looks).toBeDefined()
+      if (looks.length > 0) {
+        type = 'look'
+        id = looks[0].id!.toString(10)
+      } else {
+        const dashboards = await sdk.ok(sdk.search_dashboards({ limit: 1 }))
+        expect(dashboards).toBeDefined()
+        if (dashboards.length > 0) {
+          type = 'dashboards'
+          id = dashboards[0].id!
+        }
+      }
+      expect(type).toBeDefined()
+      expect(id).toBeDefined()
+      const image = await sdk.ok(
+        sdk.content_thumbnail({ type: type, resource_id: id, format: 'png' })
+      )
+      expect(image).toBeDefined()
+      expect(mimeType(image)).toEqual('image/png')
+      const svg = await sdk.ok(
+        sdk.content_thumbnail({ type: type, resource_id: id, format: 'svg' })
+      )
+      expect(svg).toBeDefined()
+      expect(svg).toMatch(/^<\?xml/)
+    })
+  })
+
+  describe('PUT smoke test', () => {
+    it('set default color collection', async () => {
+      const sdk = new LookerSDK(session)
+      const current = await sdk.ok(sdk.default_color_collection())
+      expect(current).toBeDefined()
+      const cols = await sdk.ok(sdk.all_color_collections())
+      const other = cols.find((c) => c.id !== current.id)
+      expect(other).toBeDefined()
+      // tests to stop lint from complaining
+      if (other && other.id && current.id) {
```

Comment:
rather than `!` everywhere? 
a/ I wish lint would recognize `expect(other).toBeDefined()` as verifying that `other` is not undefined/null
b/ dead horse but can't resist: wish our linting/typing recognized that `.id` must be non-undefined/null

all that being said, I think I'd prefer a `!` to silence the warning because reading this conditional I'm more likely to think "oh, maybe these things could actually be null?". Using `!` seems to more explicitly call out the problem we're working around?

I don't really feel strongly either way though, so ok as is

Reply:
Using the imperative makes the lint checker produce warnings. I don't like this either, but there's no other way (other than completely disabling the lint rule) to remove the warnings.